### PR TITLE
Implements function modification for InstanceManager

### PR DESCRIPTION
**Pull request by blegat**

### CODE REVIEW

File: src/instancemanager.jl [hunk 1]
```diff
@@ -207,14 +207,14 @@ function MOI.canmodifyconstraint(m::InstanceManager, cindex::CI, change)
     return true
 end
 
-function MOI.modifyconstraint!(m::InstanceManager, cindex::CI{F,S}, func::F) where {F<:MOI.AbstractFunction, S<:MOI.AbstractSet}
-    if m.mode == Automatic && m.state == AttachedSolver && !MOI.canmodifyconstraint(m.solver, cindex, typeof(func))
+function MOI.modifyconstraint!(m::InstanceManager, cindex::CI{F}, change::Union{F, MOI.AbstractFunctionModification}) where F<:MOI.AbstractFunction
```
Comment (mlubin) [inline src/instancemanager.jl]: This is a confusing method signature. What happens to `F` when `change` is a modification?

Comment (blegat) [reply src/instancemanager.jl]: `F` is determined by the type of `cindex`. Then `change` is either `F` or a function modification

Comment (mlubin) [reply src/instancemanager.jl]: The extra complexity in the type signature is probably not worth the ability to get a method error for the one corner case where `F` doesn't match the type of the function. We already check this and give a potentially more useful error message with `canmodifyconstraint`.

Comment (blegat) [reply src/instancemanager.jl]: I have simplified it

**Review by mlubin — APPROVED** [review]

**Pull request merged**